### DISCUSSION TITLE
Is it possible to have http_client run only once?

### DISCUSSION THREAD

**Discussion by tfendt-belong-gg** [redpanda-data/connect#1862] [Q&A]
I'm writing a daily task that needs to call a single api endpoint process the results and shutdown. I tried using the http_client for an input but it continuously calls the endpoint. Is there a way to have it only call once? 

I don't want to use a rate_limit as I plan on shutting down the server/container after it is complete. So I need benthos to pull from the api, process, and output to db.

The only way I found to do this was to have a dummy input:
```input:
  generate:
    mapping: "root = {}"
    count: 1

pipeline:
  processors:
    - http:
        url: "https://api-call-here.com/blah"
        verb: GET
```

This doesn't feel like the right solution...or is it?

**Comment by mihaitodor** [comment]
Hey @tfendt-belong-gg, using the `generate` input is a good solution, yes. Alternatively, you can wrap the `http_client` input in a [`read_until` input](https://www.benthos.dev/docs/components/inputs/read_until) with `check: true` and `restart_input: false`.

BTW, the `generate input` also lets you set `count: 0` along with a cron expression to the `interval` field: https://www.benthos.dev/docs/components/inputs/generate#interval, so Benthos doesn't shut down in case you ever need that.
